src/OrchardCore/OrchardCore/Environment/Shell/ShellHost.cs

```diff
@@ -408,5 +408,25 @@ private bool CanRegisterShell(ShellSettings shellSettings)
                 shellSettings.State == TenantState.Uninitialized ||
                 shellSettings.State == TenantState.Initializing;
         }
+
+
+        public void Dispose()
+        {
+            var shells = _shellContexts?.Values.ToArray();
+
+            if (shells == null || shells.Length == 0)
+            {
+                return;
+            }
+
+            foreach (var shell in shells)
+            {
+                if (shell.ServiceProvider != null)
+                {
+                    shell.Dispose();
+                    Console.WriteLine($"Disposed {shell.Settings.Name}");
```

Comment:
Oups, i forgot to remove this testing code.

For the null check it's because i was seeing the message 3 times for a given shell, so for testing i added the null check to check that a given shell need to be disposed once.

And again, i forgot to remove the check, i will update it.

Hmm, but i want to understand why shellHost dispose is called 3 times, i'm checking it.